### ISSUE TITLE
Taxonomies: ContextErrorException

### ISSUE THREAD

**Issue by peterboorsma** [bolt/bolt#7362]
There seems to be something wrong with 'group' type taxonomies. 
When I create a new taxonomy 'hoofdstukken' and add a value to a record with that taxonomy, I get this error:

```
ContextErrorException in Listing.php line 179:
Notice: Undefined offset: 1

            return $results;
        }
        if (isset($taxGroup) && $taxGroup !== null) {
            foreach ($grouped as &$group) {
                usort($group, function ($a, $b) use ($resultTaxOrders) {
                    return $resultTaxOrders[$a->getId()] - $resultTaxOrders[$b->getId()];
                });
            }
        }

        return call_user_func_array('array_merge', $grouped);

```
Bolt 3.4.9
PHP 7.1.12 / Apache
This is a new website (first version 3.4.6, updated to 3.4.9)

This is my contenttype:

```
paginas:
    name: Paginas
    singular_name: Pagina
    fields:
        locale:
            type: locale
            group: content
        slug:
            type: slug
            uses: title
        title:
            type: text
            class: large
            label: Titel
            translatable: true
	(...)
        seo:
            type: seo
            group: SEO
        nldata:
            type: hidden
        nlslug:
            type: locale_data
            index: true
        endata:
            type: hidden
        enslug:
            type: locale_data
            index: true
    recordsperpage: 100
    taxonomy: [ hoofdstukken ]
```
My taxonomy:

```
hoofdstukken:
    slug: hoofdstukken
    singular_slug: hoofdstuk
    behaves_like: grouping
    options:
        overzichtspaginas: Overzichtspagina's
        overig: Overige pagina's
    has_sortorder: true
```

**Comment by xiaohutai** [comment]
You added a grouping taxonomy AFTER there were already some records in the database?

How many records do you have?

For now, the most simple trick is to save all the records in that contenttype at least once. If you don't have a lot, then it's fairly easy. Navigate to `/bolt/editcontent/pages/1` and keep changing the `id`, andsoforth.

**Comment by peterboorsma** [comment]
That did the trick. In this case it was easy because < 10 records. 
But... Why?!

**Comment by xiaohutai** [comment]
A record (`bolt_pages`) with a `grouping` (`bolt_taxonomy`) should always have a `grouping` even if its empty. But since you added it later, they old ones are missing that taxonomy.

But also because I overlooked some things in that piece of PHP code you pasted 😅

**Comment by xiaohutai** [comment]
I made a pull request #7363 that should prevent the "error" you see in the backend.

You still need to assign a group to all ye olde ones.

**Comment by bobdenotter** [comment]
Closed by #7363.
